torch/_inductor/dependencies.py

```diff
@@ -407,21 +407,36 @@ def extract_input_node_reduction_ranges(  # noqa: F722
     reads = input_node.get_reads()
     reduction_size = None
     size = None
-    for read in reads:
-        if not isinstance(read, MemoryDep):
-            continue
-        buffer = V.graph.get_buffer(read.name)
-        if buffer is None:
-            continue
-        if isinstance(buffer, ComputedBuffer) and len(buffer.get_reduction_size()) > 0:
-            if reduction_size is None:
-                reduction_size = buffer.get_reduction_size()
-                size = buffer.get_size()
-            elif (
-                reduction_size != buffer.get_reduction_size()
-                or size != buffer.get_size()
+    seen = set()
+    while reduction_size is None and len(reads) > 0:
+        new_reads = []
+        for read in reads:
+            if not isinstance(read, MemoryDep):
+                continue
+            if read.name in seen:
+                continue
+            seen.add(read.name)
+            buffer = V.graph.get_buffer(read.name)
+            if buffer is None:
+                continue
+            if (
+                isinstance(buffer, ComputedBuffer)
+                and len(buffer.get_reduction_size()) > 0
             ):
-                return (None, None)
+                if reduction_size is None:
```

Comment:
Hmm, seems ```reduction_size``` would always be of ```None``` because of the while-loop condition above? In this case, we would never fall into the else branch.

Reply:
it's `None` at the start of the nested `for` loop, but then it gets set the first time the `for` loop finds a reduction

Reply:
Ah, that's right. I didn't noticed the inner ```for``` loop. Thanks for pointing it out.